internal/aks/create.go

```diff
@@ -159,14 +159,22 @@ func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient
 }
 
 func CreateOrUpdateAgentPool(ctx context.Context, agentPoolClient *containerservice.AgentPoolsClient, spec *aksv1.AKSClusterConfigSpec, np *aksv1.AKSNodePool) error {
+	if np.EnableAutoScaling == nil {
+		np.EnableAutoScaling = to.BoolPtr(false)
```

Comment:
Since `np` is a pointer, I think modifying it may cause confusion.

Reply:
Ok, as I have checked it wasn't needed